helios-client/src/main/java/com/spotify/helios/common/descriptors/DeploymentGroup.java

```diff
@@ -199,6 +222,9 @@ public Builder setName(final String name) {
       return this;
     }
 
+    // TODO(negz): Annotate this as @JsonProperty("job")?
+    // AFAICT we're currently serialising this with the key "jobId", despite annotating it as
+    // @JsonProperty("job") in the constructor and javadoc.
```

Comment:
It definitely serializes as `"jobId"`. 

I made a test for this and it seems like Jackson knows to modify the `jobId` field of the class after calling the constructor (through reflection somehow). So in the end, the data in the object is as expected if you translate it to json and back. 

Changing the annotations might cause incompatibilities with already stored versions 😞